engines/startrek/filestream.cpp

```diff
@@ -0,0 +1,80 @@
+#include "startrek/filestream.h"
+
+namespace StarTrek {
+
+FileStream::FileStream(Common::SeekableReadStream *stream, bool bigEndian) {
+	_bigEndian = bigEndian;
+
+	_pos = 0;
+	_size = stream->size();
+	_data = new byte[_size];
+	stream->read(_data, _size);
+	delete stream;
+}
+
+FileStream::~FileStream() {
+	delete[] _data;
+}
+
+// ReadStream functions
+
+uint32 FileStream::read(void *dataPtr, uint32 dataSize) {
```

Comment:
This was originally used like a stream, and still sometimes is, but in certain places I actually read from the byte array directly, so extending MemoryReadStreamEndian doesn't quite work. It's not purely a "stream" anymore so perhaps it should be renamed.

Reply:
You may extend MemoryReadStream() with getData() method. This is better than reimplementing the whole thing. Could you please point me to where do you read from the array directly?

Reply:
Turns out it's only it one spot, the "StarTrekEngine::getLoadedText" function, since I'm reading from it as a string (char array).

In terms of a "getData" method, it seems that I'll need to make a second copy of the file in memory to access it that way, since there seems to be no direct way to access the MemoryReadStream's data. In this particular case I could just copy the portion of the string I need, which isn't much overhead, though still technically unnecessary.

Reply:
Oh, nevermind, I can keep track of the initial byte array I pass to the MemoryReadStream. Yeah, I'll do that...